### PR TITLE
[CF-86] Use dd instead of sftp

### PR DESCRIPTION
**Pull request by SVilgelm**
In some case sftp doesn't work, to fix it use dd command by ssh
and read a data from stdout of ssh tunnel.
Implement RemoteStdout class to execute a command and work with
stdout of command.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.actions.test_recreate_boot_image
tests/cloudferrylib/os/actions/test_recreate_boot_image.py:17: [W0611(unused-import), ] Unused state imported from fabric

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2423/

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.utils.file_proxy
cloudferrylib/utils/file_proxy.py:40: [W0612(unused-variable), get_file_size] Unused variable 'e'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2426/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.utils.test_file_proxy
tests/cloudferrylib/utils/test_file_proxy.py:15: [W0611(unused-import), ] Unused import errno

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2427/

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:418: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by SVilgelm** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:418: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/utils/remote_stdout.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Copyright (c) 2014 Mirantis Inc.
```
Comment (roman-verchikov) [inline cloudferrylib/utils/remote_stdout.py]: Let's move this to `utils/files.py`

File: requirements.txt [hunk 2]
```diff
@@ -24,5 +24,4 @@ python-swiftclient==2.3.1
 pyyaml
 redis
 sqlalchemy
-# see https://github.com/paramiko/paramiko/issues/570 and https://github.com/paramiko/paramiko/pull/571
--e git://github.com/mirrorcoder/paramiko.git#egg=paramiko-1.16.0
\ No newline at end of file
+paramiko==1.16.0
```
Comment (roman-verchikov) [inline requirements.txt]: Let's remove this dependency, it's part of fabric

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:418: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by SVilgelm** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:418: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by SVilgelm** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:418: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:418: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by SVilgelm** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:418: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:418: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:418: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
Looks good, +1

File: cloudferrylib/utils/file_proxy.py [hunk 3]
```diff
@@ -38,18 +37,8 @@ def get_file_size(file_obj):
...
-                # object which doesn't support 'seek/tell' has
-                # been supplied.
-                return
-            else:
-                raise
+        except IOError:
+            return
 
 
 class ProgressView(object):
```
Comment (archyufa) [inline cloudferrylib/utils/file_proxy.py]: Please fix progress view for ReCreateBootImage file_proxy.py:69 Copying image shows incorrect values:
12.1MB of 1.8MB 665%

**Comment by archyufa** [comment]
It is not clear what ReCreateBootImage using dd or sftp. Can you make this clear from INFO logs.

**Comment by archyufa** [comment]
+1 worked in staging

File: cloudferrylib/utils/file_proxy.py (hunk 3 above)
Comment (SVilgelm) [inline cloudferrylib/utils/file_proxy.py]: Do you have a log with this message?

**Pull request merged**